libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkAuthenticationFactory.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    /// <summary>
+    /// Cloud environments capture the environment specific Bot Framework Protocol auth code.
+    /// </summary>
+    public static class BotFrameworkAuthenticationFactory
```

Comment:
The only reason for this to exist is as the most minimal entry point. I mean you need _some_ entry point.

The configuration classes (in the integration assembles use it for example.)

The goal is to have as little surface area as possible. And that was what this was intended to be: just a single function, replacing a whole bunch of public classes (which a whole bunch of public constructor on some of them.)

It's not, incidentally meant to be extensible, it's just the entry point for the implementations we provide. The actual extensibility story is the two interfaces (now abstract classes) which work at two distinct levels. (And have a dependency between them.) So you can either replace the whole implementation or just replace the actual credential creation. Anyhow that's the idea.

Do you have a suggestion as to something else we could do? The goal being minimal surface area.

And in terms of timing, the surface area we will use in the templates for the next few weeks will most likely be the configuration. This is there to enable that work. We do have an opportunity to iterate on this before we ship the next set of packages.

Reply:
We could make Internal and enable InternalsVisibleTo so you can use it in tests, I know you don't like internals visible to but it would reduce the surface area since it is, so far, only being used internally in the invisible part of the constructors. We can revisit this later if you want. Resolving it for now.